src/MakingSense.AspNetCore.Documentation/DocumentationMiddleware.cs

```diff
@@ -147,14 +165,19 @@ private string GetFilePath(string subpath, out bool isDirectory)
 				: subpath;
 		}
 
-		private bool TryGetDocumentsSubpath(HttpRequest request, out PathString subpath)
+		private bool TryGetDocumentsSubpathAndLanguage(HttpRequest request, out PathString subpath, out string lang)
 		{
+			lang = null;
+
 			if (!request.Path.StartsWithSegments(_options.RequestPath, out subpath))
 			{
 				return false;
 			}
 
-			subpath = _options.FileProviderSubPath.Add(subpath);
+			var match = _langPatternRegex.Match(subpath);
+
+			lang = match.Groups[2].Value;
+			subpath = _options.FileProviderSubPath.Add(match.Groups[3].Value);
```

Comment:
There are some scenarios where `match.Groups[3].Value` could be null, for example:

```
/es
    Group 1.    `/es`
    Group 2.    `es`
```

Do will it work fine? or should you use something like `match.Groups[3].Value ?? ""` or `match.Groups[3]?.Value ?? ""`?

Reply:
On that case, it's redirected to /es/ before evaluate the regular expression.

Anyway, on fail cases (like a string not starting with slash) it returns empty string for each group:
![image](https://user-images.githubusercontent.com/25298072/33339504-2709ce22-d458-11e7-848c-54b5bdc393e1.png)

Reply:
Nice! thanks for the detailed answer, it helps me to learn more about Regex class.